source/features/git-checkout-pr.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import ClippyIcon from 'octicon/clippy.svg';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import {getCurrentBranch} from '../github-helpers';
+
+function handleMenuOpening({delegateTarget: dropdown}: delegate.Event): void {
+	dropdown.classList.add('rgh-git-checkout'); // Mark this as processed
+	const [, user, repository] = select<HTMLAnchorElement>('.commit-ref.head-ref a')!.pathname.split('/', 3);
+	const isLocalPr = select('.user-select-contain.head-ref a')!.childElementCount === 1;
+	const tabContainer = select('[action="/users/checkout-preference"]', dropdown)!.closest<HTMLElement>('tab-container')!;
+	select('.UnderlineNav-body', tabContainer)!.append(
+		<button
+			name="type"
+			type="button"
+			role="tab"
+			aria-selected="false"
+			className="UnderlineNav-item flex-1 btn-link"
+		>
+			Git Checkout
+		</button>
+	);
+
+	tabContainer.append(
+		<div hidden role="tabpanel" className="p-3">
+			<p className="text-gray text-small">
+				Run in your project repository
+			</p>
+			<div className="copyable-terminal">
+				<div className="copyable-terminal-button">
+					<clipboard-copy
+						className="btn btn-sm zeroclipboard-button"
+						role="button"
+						for="rgh-checkout-pr"
+						aria-label="Copy to clipboard"
+						data-copy-feedback="Copied!"
+					>
+						<ClippyIcon/>
+					</clipboard-copy>
+				</div>
+				<pre
+					id="rgh-checkout-pr"
+					className="copyable-terminal-content"
+				>
+					<span className="user-select-contain">
+						{isLocalPr ? '' : `git remote add ${user} ${location.origin}/${user}/${repository}.git\ngit fetch ${user}\n`}
```

Comment:
I think. `true`/`false` won't be printed
```suggestion
						{isLocalPr || `git remote add ${user} ${location.origin}/${user}/${repository}.git`}
						{isLocalPr || `git fetch ${user}`}
```

Reply:
By the way this still doesn't work. Test your commands.

Reply:
Maybe this, but keep honing it. For example `fetch` should only fetch the branch. Or maybe there's a way to skip that command altogether

https://stackoverflow.com/a/46078150/288906

Reply:
Its was the user name in the switch.

Reply:
@fregante the current code is working

Reply:
Show the code you’re suggesting

Reply:
```shell
git remote add yakov116 https://github.com/yakov116/refined-github.git
git fetch yakov116 check-it-out
git switch check-it-out
```

Reply:
Didn't you say 2 commands?

Also that requires more commands to push back. 

```
❯ git push
fatal: The current branch check-it-out has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin check-it-out
```

The command git suggests is wrong. This works

```
git remote add yakov116 https://github.com/yakov116/refined-github.git
git fetch yakov116 check-it-out
git checkout --track yakov116/check-it-out
```

I was trying to get you to read the [documentation](https://git-scm.com/docs/git-remote) to find better commands. This I was I found:

```
git remote add yakov116 https://github.com/yakov116/refined-github.git -f -t check-it-out
git switch --track yakov116/check-it-out
```

Reply:
I missed the boat here. Sorry about that.

Reply:
> We can do it in 2 commands by grouping the remote add and the fetch in one. But if the remote exists it will error out and not do the second command.

If the remote exists then the first command which includes the fetch of the branch will fail. Then the second command will fail too with the following.
```console
git switch --track kidonng/cleanup-repo-filelist-actions
fatal: invalid reference: kidonng/cleanup-repo-filelist-actions
```

Reply:
I expected the first command to fail, but it shouldn't matter because the second command would be executed anyway.

However because the first command includes `fetch`, then also `fetch` fails, so the last command doesn't work either.

So we should keep them as 3 separate commands so that if the first one fails, the other 2 still work:

```
git remote add yakov116 https://github.com/yakov116/refined-github.git
git fetch yakov116 check-it-out
git switch --track yakov116/check-it-out
```

The only safe improvement here is if there's a way to **fetch and checkout in one command** that also includes `--track`